apps/tx-builder/src/components/Dashboard.tsx

```diff
@@ -99,6 +74,47 @@ const Dashboard = () => {
     }
   }, [sdk.txs, transactions]);
 
+  // Load contract from address or ABI
+  useEffect(() => {
+    async function loadContract(addressOrAbi: string) {
+      setContract(null);
+      setLoadContractError('');
+
+      if (!addressOrAbi || !services.web3 || !services.interfaceRepo) {
+        return;
+      }
+
+      try {
+        setIsABILoading(true);
+        const contract = await services.interfaceRepo.loadAbi(addressOrAbi);
+        setContract(contract);
+      } catch (e) {
+        setContract(null);
+        setLoadContractError('No ABI found for this address');
+        console.error(e);
+      }
+      setIsABILoading(false);
+    }
+
+    loadContract(addressOrAbi);
+  }, [addressOrAbi, services.interfaceRepo, services.web3]);
+
+  function getAddressFromDomain(name: string): Promise<string> {
+    return services?.web3?.eth.ens.getAddress(name) || new Promise((resolve) => resolve(name));
+  }
+
+  const isValidAddress = useCallback(
+    (address: string | null) => {
+      if (!address) {
+        return false;
+      }
+      return services?.web3?.utils.isAddress(address);
+    },
+    [services.web3],
+  );
+
+  const hasError = !isValidAddress(addressOrAbi) && !contract;
```

Comment:
I would call it `showBuilder` and remove the three !

Reply:
done! I called it `isValidAddressOrContract`, what is your opinion about it?